tests/ComposerJsonTest.php

```
@@ -0,0 +1,310 @@
+        $exitCode = 0;
+        exec(escapeshellarg(PHP_BINARY) . ' -l ' . escapeshellarg($tmpFile) . ' 2>&1', $output, $exitCode);
+
+        unlink($tmpFile);
+
+        $this->assertSame(
+            0,
+            $exitCode,
+            "Inline PHP script failed 'php -l' syntax validation: {$originalScript}\n" . implode("\n", $output)
+        );
+    }
+}
```

PHPMD (2.15.0)

[warning] The class ComposerJsonTest has 16 public methods. Consider refactoring ComposerJsonTest to keep number of public methods under 10. (undefined)

(TooManyPublicMethods)